### PR TITLE
Divide the project to apps - DO NOT MERGE

### PR DESCRIPTION
**Pull request by AvivLiberman**
#33

### CODE REVIEW

**Review by avraham1992 — COMMENTED** [review]
tanks bro

**Review by aradz1997 — APPROVED** [review]

**Comment by aradz1997** [comment]
We need to add the following lines to falcon/setting.py 
INSTALLED_APPS = [
    'Employees.apps.EmployeesConfig',
    'Manager.apps.ManagerConfig',
    'Task.apps.TaskConfig',
]

**Review by aradz1997 — CHANGES_REQUESTED** [review]
We need to add the following lines to falcon/setting.py
INSTALLED_APPS = [
'Employees.apps.EmployeesConfig',
'Manager.apps.ManagerConfig',
'Task.apps.TaskConfig',
]

**Review by avraham1992 — CHANGES_REQUESTED** [review]
@AvivLiberman what arad say

**Comment by orzionpour** [comment]
`# Run database migrations
pipenv run python manage.py migrate`

This line should be added to the `setup.sh` to run the migrations when using `vagrant up`

**Review by orzionpour — CHANGES_REQUESTED** [review]
`
// Run database migrations //

pipenv run python manage.py migrate
`
This line should be added to the `setup.sh` to run the migrations when using `vagrant up`

**Comment by AvivLiberman** [comment]
@aradz1997 , @orzionpour Done

**Review by orzionpour — APPROVED** [review]
LGTM

**Comment by AvivLiberman** [comment]
@mgold1234 @natifridman @nunnatsa can you please CR?

**Review by nunnatsa — COMMENTED** [review]
I think the data model is good, but I'm not sure about the split between managers an emploees for several reasons. First, the manager is also an emploee. Also, people are moving: team members become managers, moving to other teams or leaving the company. What do you think about removing the managers table, and add a "role" column to the emploee table?

**Review by avraham1992 — APPROVED** [review]

File: Task/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            name='Task',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('title', models.CharField(max_length=50)),
+                ('name_task', models.CharField(max_length=50)),
+                ('about', models.TextField(blank=True)),
+                ('status', models.CharField(max_length=50)),
+                ('publish_date', models.DateTimeField()),
+                ('property', models.IntegerField()),
+                ('author', models.ForeignKey(on_delete=django.db.models.deletion.RESTRICT, to='Manager.manager')),
```
[the remark this answers is not in the document]
Comment (avraham1992) [reply Task/migrations/0001_initial.py:26]: So you propose to make the manager an employee with permission to write tasks? I only have if it's a problem of the foreign key once a manager leaves must change all the tasks of the employees to the tasks of a new manager...

[the remark this answers is not in the document]
Comment (avraham1992) [reply Task/migrations/0001_initial.py:26]: from django.db import models
from django.contrib.auth.models import User


class Profile(models.Model):
    user = models.OneToOneField(User, on_delete=models.CASCADE)
    image = models.ImageField(default='default.jpg', upload_to='profile_pics')
    dob = models.DateField(blank=True, null=True)

    def __str__(self):
        return f'{self.user.username} Profile'

    def save(self, *args, **kwargs):
        super().save(*args, **kwargs)

[the remark this answers is not in the document]
Comment (avraham1992) [reply Task/migrations/0001_initial.py:26]: `from django.db import models
from django.db.models.deletion import CASCADE
from User.models import Profile

class Task(models.Model):
    author = models.ForeignKey(Profile, on_delete=models.RESTRICT)
    title = models.CharField(max_length=50)
    name_task = models.CharField(max_length=50)
    about = models.TextField(blank=True)
    status = models.CharField(max_length=50)
    publish_date = models.DateTimeField()
    property = models.IntegerField()`

[the remark this answers is not in the document]
Comment (avraham1992) [reply Task/migrations/0001_initial.py:26]: this Okey
?

[the remark this answers is not in the document]
Comment (avraham1992) [reply Task/migrations/0001_initial.py:26]: Basically I went for the thinking that it's the manager's account so once he leaves he does not have to keep the task board that he no longer deals with pops up...

**Comment by nunnatsa** [comment]
@AvivLiberman please edit the PR description. Please explain what this PR do, and why is it needed. Description like `#33` won't help the reviewers, and more import, won't help you in the future when tou'll try to remember what was changed in this PR and why it was done, while trying to debug an issue on the customer site, in the middle of the night, on weekend.

File: Task/migrations/0001_initial.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mgold1234) [reply Task/migrations/0001_initial.py:26]: but if the manager will move to a different role in the same company? we want to keep the task board for him

**Comment by AvivLiberman** [comment]
@nunnatsa it wasn't my PR - I have opened it because @avraham1992 got a mess one

**Comment by mgold1234** [comment]
@AvivLiberman, but for someone that open this PR #33 is not clear

**Comment by AvivLiberman** [comment]
@mgold1234 you are right, I think that this PR include 2 tasks in it:

1.  #25 
2.  #23 

And I don't think the structure is good. I will open new one

**Comment by mgold1234** [comment]
@AvivLiberman , why do you think that the structure is not good?

**Pull request closed without merging**